CUAir/nimbul
# Adding Websocket Server for telemetry and status

## Summary 
Adding a WebSocket server in MAVProxy instead of a REST API for low latency and small overhead update to the frontend.
Sharing a "message" queue across threads to constantly send data to the frontend

## Testing
1. Run SITL (Tested on Mac):
```docker run -itd --rm --name sitl_container cuairautopilot/private_repo:sitl_image -C --out=udp:host.docker.internal:14550```

2. Run MAVProxy (From root of the repo):
```python -m MAVProxy.mavproxy --sitl=sitl_container:14550```

3. Run frontend (From frontend folder)
```npm run start```

See Logs on the terminal.

## Notes
Still need to support the different kinds of messages
